### PR TITLE
feat(plugin-conversation): add options param to download method

### PR DESCRIPTION
**Pull request by aimeex3**
# Pull Request Template

## Description

Add options param to the download method to allow custom headers for the GET request. For example, these requests are not needed in the giphy request 
```
headers: {
        'cisco-no-http-redirect': null,
        'spark-user-agent': null,
        trackingid: null
      }
```
It throws a CORS error if these are passed in FF

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] From web client, pass in options with the above headers in the description. No CORS error in FF

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@ciscospark/internal-plugin-encryption/src/encryption.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ const Encryption = SparkPlugin.extend({
         .then((result) => result.plaintext.toString()));
   },
 
-  download(scr) {
+  download(scr, options) {
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/internal-plugin-encryption/src/encryption.js]: Could you split this out to another commit?

Comment (aimeex3) [reply packages/node_modules/@ciscospark/internal-plugin-encryption/src/encryption.js]: actually didn't need to add options to the encryption plugin. removed from PR and updated tests

File: packages/node_modules/@ciscospark/internal-plugin-conversation/src/conversation.js [hunk 2]
```diff
@@ -143,12 +143,13 @@ const Conversation = SparkPlugin.extend({
    * @param {Object} item
    * @param {Object} item.scr
    * @param {string} item.url
+   * @param {Object} options
```
Comment (mattnorris) [inline packages/node_modules/@ciscospark/internal-plugin-conversation/src/conversation.js]: Rather than an object, please enumerate some expectations, e.g., `options.headers`

Comment (aimeex3) [reply packages/node_modules/@ciscospark/internal-plugin-conversation/src/conversation.js]: 👍 added options.headers

**Comment by mattnorris** [comment]
@aimeex3 would you still like to merge this?

**Comment by aimeex3** [comment]
i think it was already merged
